apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.IntegrationTests/SimplePubSubTest.cs

```diff
@@ -224,5 +224,39 @@ public SimplePubSubTest(PubsubFixture fixture)
             }
             await simplePublisher.ShutdownAsync(TimeSpan.FromSeconds(15)).ConfigureAwait(false);
         }
+
+        [Fact]
+        public async Task MaxBatchSize()
+        {
+            // Test sending messages such that the maximum batch size is reached.
+            var topicId = _fixture.CreateTopicId();
+            // Create topic
+            var topicName = new TopicName(_fixture.ProjectId, topicId);
+            var publisher = await PublisherClient.CreateAsync().ConfigureAwait(false);
+            await publisher.CreateTopicAsync(topicName).ConfigureAwait(false);
+            // Create SimplePublisher, with no retry
+            var batchingSettings = new BatchingSettings(SimplePublisher.ApiMaxBatchingSettings.ElementCountThreshold,
+                SimplePublisher.ApiMaxBatchingSettings.ByteCountThreshold, TimeSpan.FromSeconds(4));
```

Comment:
Is 4 seconds long enough for 10MB?

Reply:
Yes, 4 seconds is for locally filling the batch, not for sending over the network.